### ISSUE TITLE
Hey there, good looking

### ISSUE THREAD

**Issue by deltaidea** [twestura/language-aoe2-rms#1]
Hi!
You implemented the same stuff as I did because I was a lazy bastard and didn't tell anyone anywhere about my AoE-related projects.

Syntax highlighting: https://github.com/mangudai/sublime-text ([also for VS Code](https://github.com/mangudai/vscode/tree/master/client))

Editor-agnostic parser + linter + autocomplete: https://github.com/mangudai/mangudai (unfinished, I had to abandon it last year because of work)

Let's join forces! Would you be interested in merging our syntax definitions and working on IDE tooling together? These repos above haven't been maintained for quite some time but I could revive them in a week.

I've found you by searching "aoe2" on GitHub. I'm really glad that someone else wants such utilities!

---

_Edit:_ I didn't realize you're _the_ TWest! That's so cool! 😄 You may have seen me as DontForgetToBeAwesome or deltaidea on Twitch but if not, doesn't matter.

**Comment by twestura** [comment]
Hello!
Was going to continue working on this in about a month or so. I wanted first to make an updated version of [Zetnus's rms scripting guide](http://forums.heavengames.com/redir/http%3A//aok.heavengames.com/blacksmith/showfile.php%3Ffileid%3D12178) so that it's viewable in a nicely formatted webpage, not just a Word document 11. As part of that I'd like to construct documentation similar to what is present in the UP reference documentation. And then have some type of documentation that pops up in your editor when hovering over commands in the rms.

There are a couple of other tools I was thinking of making, such as automating the construction of a map pack from multiple rms scripts, but I put that off after discovering some bugs in aoe2's rms parser 11. The UP team can of course fix bugs as they're discovered, but HD will always remain broken (although I don't know how worthwhile it is to maintain support for the HD version).

Anyway, at some point soon I will write up a more formal design of what I plan to implement, then hopefully in a month or so will have time to work on it. Would be great to put together a set of editor tools to support sublime+vs code+atom+emacs+vim+whatever other editors people use.

**Comment by deltaidea** [comment]
This vaguely aligns with what I had in mind as well!

Where can I reach map scripters and other developers like you? Would be good to discuss this with the people who are actually going to use all this and also to solidify knowledge about the game parser.

**Comment by twestura** [comment]
> Where can I reach map scripters and other developers

[AoEZone](https://www.aoezone.net/), [AoKHeaven](http://aok.heavengames.com/), and the [AI Scripters forum](http://forums.aiscripters.com/).

**Comment by deltaidea** [comment]
Yep, these links are extremely helpful! Also T90's Discord. If anyone else is reading this: go to https://www.twitch.tv/t90official and type `!discord` in the chat.
